File: camera/src/main/java/com/dastanapps/camera/listeners/Cam1OrientationEventListener.java
package com.dastanapps.camera.listeners;

import android.content.Context;
import android.hardware.SensorManager;
import android.util.Log;
import android.view.Surface;

/**
 * Created by dastaniqbal on 19/01/2018.
 * dastanIqbal@marvelmedia.com
 * 19/01/2018 11:08
 */

public class Cam1OrientationEventListener extends android.view.OrientationEventListener {
    private int mOrientation;

    public Cam1OrientationEventListener(Context context) {
        super(context, SensorManager.SENSOR_DELAY_NORMAL);
    }

    @Override
    public void onOrientationChanged(int orientation) {
        int lastOrientation = mOrientation;

        if (orientation >= 315 || orientation < 45) {
            if (mOrientation != Surface.ROTATION_0) {
                mOrientation = Surface.ROTATION_0;
            }
        } else if (orientation >= 45 && orientation < 135) {
            if (mOrientation != Surface.ROTATION_90) {
                mOrientation = Surface.ROTATION_90;
            }
        } else if (orientation >= 135 && orientation < 225) {
            if (mOrientation != Surface.ROTATION_180) {
                mOrientation = Surface.ROTATION_180;
            }
        } else if (mOrientation != Surface.ROTATION_270) {
            mOrientation = Surface.ROTATION_270;
        }

        if (lastOrientation != mOrientation) {
            Log.d("!!!!", "rotation!!! lastOrientation:"
                    + lastOrientation + " mOrientation:"
                    + mOrientation + " orientaion:"
                    + orientation);
        }
    }
}

File: camera/src/main/java/com/dastanapps/camera/listeners/Cam1OrientationEventListener.kt
package com.dastanapps.camera.listeners

import android.app.Activity
import android.content.Context
import android.hardware.SensorManager
import android.os.Handler
import android.util.Log
import android.view.OrientationEventListener
import com.dastanapps.CommonUtils
import com.dastanapps.view.Util

/**
 * Created by dastaniqbal on 19/01/2018.
 * dastanIqbal@marvelmedia.com
 * 19/01/2018 11:08
 */

class Cam1OrientationEventListener(val context: Context, val handler: Handler) : android.view.OrientationEventListener(context, SensorManager.SENSOR_DELAY_NORMAL) {
    private var mOrientation: Int = 0
    private var mOrientationCompensation: Int = 0

    override fun onOrientationChanged(orientation: Int) {
        // We keep the last known orientation. So if the user first orient
        // the camera then point the camera to floor or sky, we still have
        // the correct orientation.
        if (orientation == OrientationEventListener.ORIENTATION_UNKNOWN) return
        mOrientation = Util.roundOrientation(orientation, mOrientation)
        // When the screen is unlocked, display rotation may change. Always
        // calculate the up-to-date orientationCompensation.
        val orientationCompensation = mOrientation + Util.getDisplayRotation(context as Activity?)
        if (mOrientationCompensation != orientationCompensation) {
            mOrientationCompensation = orientationCompensation
        }

        Log.d("!!!!", "rotation!!! lastOrientation: $orientation mOrientationCompensation: $mOrientationCompensation")
        CommonUtils.sendMessageToHandler(handler, 3, mOrientationCompensation)
    }
}
